src/components/structures/auth/SetupEncryptionBody.js

```diff
@@ -39,13 +40,24 @@ function keyHasPassphrase(keyInfo) {
     );
 }
 
+// Maximum acceptable size of a key file. It's 59 characters including the spaces we encode,
+// so this should be plenty and allow for people putting extra whitespace in the file because
+// maybe that's a thing people would do?
+const KEY_FILE_MAX_SIZE = 128;
+
+// Don't shout at the user that their key is invalid every time they type a key: wait a short time
+const VALIDATION_THROTTLE_MS = 200;
+
 export default class SetupEncryptionBody extends React.Component {
     static propTypes = {
         onFinished: PropTypes.func.isRequired,
     };
 
     constructor() {
         super();
+
+        this._fileUpload = null;
```

Comment:
can we use createRef instead of manual ref collectors plz

Reply:
Mmm, good point, that's much better.